odl/tomo/geometry/conebeam.py

```diff
@@ -624,6 +626,13 @@ def __init__(self, apart, dpart, src_radius, det_radius, pitch=0,
         det_radius : nonnegative float
             Radius of the detector circle. Must be nonzero if ``src_radius``
             is zero.
+        det_curvature_radius : nonnegative float or a sequence of
+            two nonnegative floats, optional
+            Radius or radii of the detector curvature.
+            If ``None``, a flat detector is used,
+            if one value, a cylindrical detector is used,
+            if a sequence of two elements, e.g. [r, r], a spherical detector
+            is used.
```

Comment:
I have implemented only r1=r2. It seems to be very easy to extend it given the formulas that you just have proposed. 
However, from practical point of view this is not yet needed.

Reply:
I agree, it seems like an uncommon case. Interface-wise I think we should offer the full flexibility so that this case can be added later. Currently we can just raise a `NotImplementedError`.